docs/interfaces/datastore-service.txt

```diff
@@ -84,14 +84,28 @@ Interface CreateEvent {
     }
 }
 
-// Note: For deleting keys, they must be set to `None`. These keys
-// will be removed from the model.
+// Note: For deleting keys, they must be set to `None`. These keys will be removed from
+// the model.
+// list_fields can be used to partially update list fields: the values in `add` will be
+// appended to the given field, the values in `remove` will be removed from the field.
+// The case of double adding or removing something invalid will be silently ignored.
+// All given list_fields must be flat arrays of strings or ints, else an error is
+// thrown.
+// Either fields or list_fields must be given or an error will be thrown.
```

Comment:
I miss information about different behaviours:

- fields are given, that are not lists
- fields that are lists, but during comparisons the content are not strings or integers
- adding to a non-existing
- Removing from an empty list.